client/components/domain-diagnostic/index.jsx

```diff
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+import Notice from 'components/notice';
+import NoticeAction from 'components/notice/notice-action';
+
+export default React.createClass( {
+
+	displayName: 'DomainDiagnostic',
+
+	propTypes: {
+		domain: React.PropTypes.string.isRequired,
+	},
+	getInitialState: function() {
+		return {
+			domainDidFail: false
+		};
+	},
+
+	componentDidMount() {
+		const self = this;
+		const imageChecker = new Image();
+		imageChecker.onerror = function() {
+			self.setState( { domainDidFail: true } );
+		}
+		imageChecker.src = 'http://' + this.props.domain + '/w32.gif';
```

Comment:
We currently have an endpoint for this: `/domains/%s/check-nameservers`. I think it's way better to do a server-side check. Downside is that you'd have to query for each domain.

We also have something similar to this: https://github.com/Automattic/wp-calypso/pull/3329
It's waiting for me to get some data.